src/globals.css

```diff
@@ -92,6 +92,23 @@ samp {
 	}
 }
 
+/* Previewed code block with side-by-side layout */
+.previewed-code {
+	@apply rounded-md overflow-hidden flex flex-col lg:flex-row lg:flex-wrap;
+}
+
+.previewed-code pre {
+	@apply flex-1 m-0 rounded-none whitespace-pre lg:flex-[1_0_auto];
```

Comment:
After adding `whitespace-pre`, the `<pre>` may cause the page to overflow horizontally  on narrow screens.

Example: ([en-US v0.14.0 `/reference/styling/`](https://deploy-preview-27--typst-docs-web.netlify.app/en-us-v0.14.0/reference/styling/#show-rules), `window.innerWidth` = 450)

<img width="400" alt="图片" src="https://github.com/user-attachments/assets/b2d28455-eb66-49ae-809d-810cb0791380" />

```suggestion
	@apply flex-1 m-0 rounded-none whitespace-pre overflow-x-auto lg:flex-[1_0_auto];
```

https://github.com/typst-community/typst-docs-web/blob/2ae7f83e4f27bd53a9c0cb5edcf7ee1b72f8610b/src/components/templates/BaseTemplate.tsx#L179

Adding `overflow-x-auto` here, and changing this `flex` in `<BaseTemplate>` to `lg:flex` would make `<pre>` scrollable and avoid the page from overflowing horizontally.

<img width="400" alt="图片" src="https://github.com/user-attachments/assets/33e779a4-ef05-4f77-bda1-f8a3fa05cc2a" />